libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/RegressionDistributionChart.tsx

```diff
@@ -0,0 +1,306 @@
+
+  public constructor(props: IRegressionDistributionChartProps) {
+    super(props);
+    this.state = { targetFlyoutIsVisible: false };
+  }
+
+  public componentDidMount(): void {
+    if (this.state.targetOption === undefined) {
+      const firstOption = targetOptions[0];
+      this.setState({
+        newlySelectedTargetOption: firstOption,
+        targetOption: firstOption
+      });
+    }
+  }
+
+  public async componentDidUpdate(
+    prevProps: IRegressionDistributionChartProps,
+    prevState: IRegressionDistributionChartState
+  ): Promise<void> {
+    if (
+      this.props.boxPlotState.boxPlotData.length === 0 ||
+      !_.isEqual(prevProps.cohorts, this.props.cohorts) ||
+      prevState.targetOption !== this.state.targetOption
+    ) {
+      const targetOption = this.state.targetOption
+        ? this.state.targetOption?.key.toString()
+        : targetOptions[0].key.toString();
+      const boxPlotData = await Promise.all(
+        this.props.cohorts.map((cohort: ErrorCohort, index: number) => {
+          return calculateBoxPlotDataFromErrorCohort(
+            cohort,
+            index,
+            targetOption
+          );
+        })
+      );
+      const outlierData = boxPlotData
+        .map((cohortBoxPlotData) => cohortBoxPlotData?.outliers)
+        .map((outlierValues, cohortIndex) => {
+          return outlierValues?.map((val) => [cohortIndex, val]);
+        })
+        .filter((list) => list !== undefined)
+        .reduce((list1, list2) => {
+          if (list1 === undefined) {
+            return list2 ?? [];
+          }
+          if (list2 === undefined) {
+            return list1 ?? [];
+          }
+          return list1.concat(list2);
+        }, []);
+      if (
+        !_.isEqual(boxPlotData, this.props.boxPlotState.boxPlotData) ||
+        !_.isEqual(this.props.boxPlotState.outlierData, outlierData)
+      ) {
+        this.props.onBoxPlotStateUpdate({ boxPlotData, outlierData });
+      }
+    }
+  }
+
+  public render(): React.ReactNode {
+    const theme = getTheme();
+    const classNames = modelOverviewChartStyles();
+    if (
+      !this.context.jointDataset.hasTrueY ||
+      !this.context.jointDataset.hasPredictedY
+    ) {
+      return;
+    }
+
+    if (targetOptions.length === 0) {
+      return React.Fragment;
+    }
+
+    if (this.state.targetOption === undefined) {
+      this.setState({ targetOption: targetOptions[0] });
+      return React.Fragment;
+    }
+
+    const noCohortSelected = this.props.cohorts.length === 0;
+
+    const selectedCohortNames = this.props.cohorts.map(
+      (cohort) => cohort.cohort.name
+    );
+
+    return (
+      <Stack
+        tokens={{ childrenGap: "10px" }}
+        id="modelOverviewRegressionDistributionChart"
+      >
+        {!noCohortSelected && (
+          <Stack.Item className={classNames.cohortSelectionButton}>
+            <DefaultButton
+              id="modelOverviewRegressionDistributionBoxChartCohortSelectionButton"
```

Comment:
that's some descriptive id  :laughing:

Reply:
You're right, of course, and I dislike it for its length, but I try to err on the side of being descriptive. Coming back to this after a few months I appreciate that it's got all the information on there. 🙂